vocode/streaming/streaming_conversation.py

```diff
@@ -46,13 +46,248 @@
+                    )
+                    response = None
+                    should_stop = True
+                # if self.conversation.agent.get_agent_config().send_filler_audio:
+                #     self.conversation.interrupt_all_synthesis()
+                #     await self.conversation.wait_for_filler_audio_to_finish()
+                if should_stop:
+                    self.conversation.logger.debug("Agent requested to stop")
+                    self.conversation.mark_terminated()
+                    return
+                if response:
+                    event = self.conversation.enqueue_interruptible_event(
+                        payload=BaseMessage(text=response),
+                        is_interruptible=self.conversation.agent.get_agent_config().allow_agent_to_be_cut_off,
+                    )
+                    self.output_queue.put_nowait(event)
+                else:
+                    self.conversation.logger.debug("No response generated")
+            if goodbye_detected_task:
+                try:
+                    goodbye_detected = await asyncio.wait_for(
+                        goodbye_detected_task, 0.1
+                    )
+                    if goodbye_detected:
+                        self.conversation.logger.debug(
+                            "Goodbye detected, ending conversation"
+                        )
+                        self.conversation.mark_terminated()
+                        return
+                except asyncio.TimeoutError:
+                    self.conversation.logger.debug("Goodbye detection timed out")
+
+    class AgentResponsesWorker(InterruptibleWorker):
+        def __init__(
+            self,
+            input_queue: asyncio.Queue[InterruptibleEvent[BaseMessage]],
+            output_queue: asyncio.Queue[SynthesisResult],
+            conversation: "StreamingConversation",
+        ):
+            self.input_queue = input_queue
+            self.output_queue = output_queue
+            self.conversation = conversation
+            self.chunk_size = (
+                get_chunk_size_per_second(
+                    self.conversation.synthesizer.get_synthesizer_config().audio_encoding,
+                    self.conversation.synthesizer.get_synthesizer_config().sampling_rate,
+                )
+                * TEXT_TO_SPEECH_CHUNK_SIZE_SECONDS
+            )
+
+        async def process(self, item: InterruptibleEvent[BaseMessage]):
+            agent_response = item.payload
+            self.conversation.logger.debug("Synthesizing speech for message")
+            self.conversation.current_synthesis_span = tracer.start_span(
+                SYNTHESIS_TRACE_NAME,
+                {
+                    "synthesizer": str(
+                        self.conversation.synthesizer.get_synthesizer_config().type
+                    )
+                },
+            )
+            synthesis_result = await self.conversation.synthesizer.create_speech(
```

Comment:
As discussed, if the synthesis_result async_generator is returned by another a synthesizer that runs in a thread, there will be thread issues, like the generator not being called as new result is added. Maybe a put a todo to address that in the future?